### PR TITLE
Improved content description of binaries and resources dirs (#1822)

### PR DESCRIPTION
**Pull request by friedrichatgc**
Clearly define which files should go the binaries and resources dir of the FMU which when these dirs must be unpacked by the FMU importer.

### CODE REVIEW

**Review by andreas-junghanns — APPROVED** [review]

File: docs/2_5_fmu_distribution.adoc [hunk 1]
```diff
@@ -84,19 +87,20 @@ The header files `fmi3PlatformTypes.h`, `fmi3FunctionTypes.h` and `fmi3Functions
 
 ===== Directory `binaries` [[binaries-directory]]
 
-A binary FMU must contain the binary files for all supported platforms in this folder.
+A binary FMU must contain, for each supported platform, the shared library `<modelIdentifier>.{dll|dylib|so}` and all files which are needed to load this shared library
```
Comment (t-sommer) [inline docs/2_5_fmu_distribution.adoc:90]: An FMU may  also ship with static libraries.

**Review by t-sommer — CHANGES_REQUESTED** [review]

**Comment by chrbertsch** [comment]
Webmeeting: @t-sommer : I will come up with proposals to simplify the wording. Could also be backported to FMI 2.0.x

File: docs/2_5_fmu_distribution.adoc [hunk 2]
```diff
@@ -87,11 +88,23 @@ The header files `fmi3PlatformTypes.h`, `fmi3FunctionTypes.h` and `fmi3Functions
...
 The FMU importer must unpack these files prior to loading the shared library `<modelIdentifier>.{dll|dylib|so}` and these files must stay there until unloading the shared library is finished.
 
+====== FMUs with static libraries [[binaries-directory-static]]
+
+A static library binary FMU must contain, for each supported platform and ABI, the static library `<modelIdentifier>.{a|lib}` and all files which are needed to link with this static library
+in the corresponding platform-specific binary folder.
+These binary files may go into an ABI specific subdirectory, see <<platform-tupe-definition>> also for linking instructions.
+For static libraries the `binaries` directory can only contain files needed during link time since, in contrast to shared libraries, it is not possible to determine the file location
+of the static library at load time.
+Hence, the FMU importer is not needed to unpack these files during execution of the FMU.
```
Comment (pmai) [inline docs/2_5_fmu_distribution.adoc:106]:
```suggestion
Hence, the FMU importer is not required to unpack these files during execution of the FMU.
```

**Review by pmai — COMMENTED** [review]
Content-wise seems the right thing to me, tightened wording would be OK by me.

File: docs/2_5_fmu_distribution.adoc [hunk 3]
```diff
@@ -84,19 +88,32 @@ The header files `fmi3PlatformTypes.h`, `fmi3FunctionTypes.h` and `fmi3Functions
 
 ===== Directory `binaries` [[binaries-directory]]
 
-A binary FMU must contain the binary files for all supported platforms in this folder.
+====== FMUs with shared libraries [[binaries-directory-shared]]
+
+A shared library binary FMU must contain, for each supported platform, the shared library `<modelIdentifier>.{dll|dylib|so}` and all files which are needed to load this shared library
+in the corresponding platform-specific binary folder.
 To use the binaries of a specific platform, all items in that platform-specific binary folder must be unpacked at the same location as the binary `<modelIdentifier>.{dll|dylib|so}`.
+This is needed since the shared library may search for other files, needed during library load time, relative to its own file location.
```
Comment (t-sommer) [inline docs/2_5_fmu_distribution.adoc:96]: This reasoning should not be normative.

File: docs/2_5_fmu_distribution.adoc [hunk 4]
```diff
@@ -84,19 +88,32 @@ The header files `fmi3PlatformTypes.h`, `fmi3FunctionTypes.h` and `fmi3Functions
...
+The FMU importer must unpack these files prior to loading the shared library `<modelIdentifier>.{dll|dylib|so}` and these files must stay there until unloading the shared library is finished.
+
+====== FMUs with static libraries [[binaries-directory-static]]
+
+A static library binary FMU must contain, for each supported platform and ABI, the static library `<modelIdentifier>.{a|lib}` and all files which are needed to link with this static library
```
Comment (t-sommer) [inline docs/2_5_fmu_distribution.adoc:101]: Requiring "all files which are needed to link with this static library" to be contained in the FMU would render any FMU illegal that has external dependencies (caused by the static library).

File: docs/2_5_fmu_distribution.adoc [hunk 5]
```diff
@@ -84,19 +88,32 @@ The header files `fmi3PlatformTypes.h`, `fmi3FunctionTypes.h` and `fmi3Functions
...
+in the corresponding platform-specific binary folder.
+These binary files may go into an ABI specific subdirectory, see <<platform-tupe-definition>> also for linking instructions.
```
Comment (t-sommer) [inline docs/2_5_fmu_distribution.adoc:103]: This folder must not be a subdirectory of the platform binary directory but the `binaries` directory. Also, we should not allow spreading libraries and dependencies across multiple folders.

**Review by t-sommer — CHANGES_REQUESTED** [review]

File: docs/2_5_fmu_distribution.adoc (hunk 5 above)
Comment (t-sommer) [reply docs/2_5_fmu_distribution.adoc:103]: IMHO, we should have a concise set of rules for the naming and availability at runtime in the spec and move all the reasoning and explanations to the implementers guide.

**Comment by andreas-junghanns** [comment]
All: Markus dropped off our planet. Any change we want to be made, we must now make. @t-sommer : will you take over? You seem most interested...

**Comment by chrbertsch** [comment]
FMI Design Meeting: 
Pierre: the original intent was to clarify that in binaries only things are put that are needed at link or load time.
For things needed at runtime, they have to be put in resources
Torsten: Give a rational that the FMU may depend on files that must be present during load time. 
Pierre: Link time is different for statically or dynamically linked library.
Pierre: There is a whole section in the IG! The reason for this PR was, that people were unclear that platform specific files can be in resources.
Pierre: Similar questions currently arise in the XCP LS, so a clarification is welcome. E.g. would not expect A2L files in resources or binaries, as they are not needed during runtime. Only for specific use cases the A2L file would be needed during runtime ...

Christian: I will create a new PR

**Comment by chrbertsch** [comment]
Superseded by #1849
